drv/gimlet-seq-server/src/seq_spi.rs

```diff
@@ -142,6 +142,10 @@ impl<S: SpiServer> SequencerFpga<S> {
         let header = CmdHeader { cmd, addr };
         let header = header.as_bytes();
 
+        if data_out.len() > data.len() - header.len() {
+            return Err(spi_api::SpiError::BadTransferSize);
+        }
```

Comment:
Does this check make the conditional below at L154 superfluous?

Reply:
Yes -- changed to an `assert!`